### PR TITLE
Update ExpectationSimulationAlgorithm doc

### PR DESCRIPTION
**Pull request by mbaudin47**

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -109,6 +112,7 @@
  * #1787 (CovarianceMatrix from SymmetricMatrix raises InvalidArgumentException)
  * #1790 (LinearModelResult.getFormula() method is not updated by Stepwise regression)
  * #1794 (Some doc examples seem to behave with new infrastructure)
+<<<<<<< HEAD
```
Comment (sofianehaddad) [inline ChangeLog:115]:
```suggestion
```

File: ChangeLog [hunk 2]
```diff
@@ -119,6 +123,8 @@
  * #1835 (RegularizedIncompleteBeta returns nan)
  * #1836 (Hypergeometric results differ without boost)
  * #1837 (Poor performance when using pickle on OT objects)
+=======
```
Comment (sofianehaddad) [inline ChangeLog:126]:
```suggestion
```

File: ChangeLog [hunk 3]
```diff
@@ -119,6 +123,8 @@
...
+>>>>>>> 6a08c679e... ExpectationSimulationAlgorithm: Fixes the doc.
```
Comment (sofianehaddad) [inline ChangeLog:127]:
```suggestion
```

**Review by sofianehaddad — COMMENTED** [review]
Please clean your patch

File: ChangeLog (hunk 1 above)
Comment (mbaudin47) [reply ChangeLog:115]: Oups. Fixed.

File: ChangeLog (hunk 3 above)
Comment (mbaudin47) [reply ChangeLog:127]: Correct. Fixed.

File: python/src/ExpectationSimulationAlgorithm_doc.i.in [hunk 4]
```diff
@@ -11,70 +11,281 @@ X : :class:`~openturns.RandomVector`
...
+Hence, both the mean and the variance must be finite. 
+For example, consider the Student distribution with :math:`\nu` degrees of freedom. 
+If :math:`\nu=1` (where the expectation is undefined) 
+or :math:`\nu=2` (where the variance is undefined), 
+then the algorithm cannot succeed. 
+
+This algorithm estimates :math:`\Expect{\vect{X}}` based on a trade-off between 
+performance and accuracy, using the smallest possible sample 
+(blockwise) size which satisfies pre-defined stopping rules. 
+It increases the sample size :math:`n` incrementally until a stopping criteria is met.
```
[the remark this answers is not in the document]
Comment (mbaudin47) [reply python/src/ExpectationSimulationAlgorithm_doc.i.in:26]: You are right. To fix this, I just remove the duplicated text.

File: python/src/ExpectationSimulationAlgorithm_doc.i.in [hunk 5]
```diff
@@ -11,70 +11,281 @@ X : :class:`~openturns.RandomVector`
...
+is successively equal to 100, 200, etc... 
+Each block of evaluation of the outputs can be parallelized, which may improve the performance of the algorithm.
+We suggest to set the block size as a multiple of the number of cores. 
+
+The algorithm stops either when the maximum number of outer iterations is reached or when the target precision is met. 
+The latter stopping criterion is based on either the coefficient of variation of the 
+sample mean (relative criterion) or the standard deviation of the sample 
+mean (absolute criterion). 
+
+Let :math:`\vect{X}\in\Rset^{n_X}` be the random vector.
```
[the remark this answers is not in the document]
Comment (mbaudin47) [reply python/src/ExpectationSimulationAlgorithm_doc.i.in:48]: Correct. I move the definition of X at the top of the text, so that it is clearer.

**Comment by mbaudin47** [comment]
The significant parts of this PR were already merged at https://github.com/openturns/openturns/pull/1717. The remaining details are fixed in this PR.

**Review by josephmure — APPROVED** [review]

**Pull request merged**